src/AutoMapper/QueryableExtensions/ExpressionRequest.cs

```diff
@@ -54,12 +54,12 @@ public override bool Equals(object obj)
 
         public override int GetHashCode()
         {
-            unchecked
-            {
-                var hashCode = SourceType.GetHashCode();
-                hashCode = (hashCode*397) ^ DestinationType.GetHashCode();
-                return MembersToExpand.Aggregate(hashCode, (currentHash, p) => (currentHash * 397) ^ p.GetHashCode());
+            var hashCode = HashCodeCombiner.Combine(SourceType, DestinationType);
+            foreach(var member in MembersToExpand)
+            {
+                hashCode = HashCodeCombiner.CombineCodes(hashCode, member.GetHashCode());
             }
+            return hashCode;
```

Comment:
```
public override int GetHashCode() => HashCodeCombiner.Combine(SourceType, DestinationType, MembersToExpand);
```
as per refactoring note.

Also can cache, but I don't think you get much performance gain out of it.